command.py

```diff
@@ -0,0 +1,310 @@
+            metavar='FILE',
+            nargs='*',
+            help="Name or ID of the file to Download. If not specified, will try as ID then as Name."
+        )
+        name_or_id = parser.add_mutually_exclusive_group()
+        name_or_id.add_argument(
+            '-n',
+            '--name',
+            help='Will use FILE as a filename to search',
+            action='store_true'
+        )
+        name_or_id.add_argument(
+            '-i',
+            '--id',
+            help='Will use FILE as a file ID to download',
+            action='store_true'
+        )
+        parser.add_argument(
+            '-l',
+            '--last',
+            help="Downloads the last modified file",
+            action='store_true'
+        )
+        parser.add_argument(
+            '-e',
+            '--extract',
+            help='Tries to extract the downloaded file',
+            action='store_true'
+        )
+
+    async def execute(self):
+        extract = self.args.extract
+        if self.args.last:
+            await self.download_last_uploaded_file(extract)
+        elif not self.args.file:
+            print('Please inform the ID or name of the file you want to download. Exiting...')
+            return
+        elif self.args.name:
+            await self.download_filename(extract)
+        elif self.args.id:
+            await self.download_id(extract)
+        else:
+            await self.try_download_id_then_name(extract)
+
+    async def download_last_uploaded_file(self, extract):
+        last_file = self.google.get_last_modified_file()
+        if not last_file:
+            print("Could not find any file")
+            return
+        await self.download_from_metadata(last_file, extract)
+
+    async def download_filename(self, extract):
+        file_found, _ = self.google.search_filename(self.file)
+        if not file_found:
+            print("Could not find any file that contains '%s' on the name" % self.file)
+            return
+        return await self.download_from_metadata(file_found, extract)
+
+    async def download_id(self, extract):
+        file_found = self.google.get_file_metadata(self.file)
+        if not file_found:
+            print("Could not find file with '%s' as ID" % self.file)
+            return
+        return await self.download_from_metadata(file_found, extract)
+
+    async def try_download_id_then_name(self, extract):
+        file_id_found = self.google.get_file_metadata(self.file)
+        if file_id_found:
+            return await self.download_from_metadata(file_id_found, extract)
+        return await self.download_filename(extract)
+
+    async def download_from_metadata(self, metadata, extract):
+        if metadata is None:
+            print('Could not find file to download')
+            return
+        file_name = metadata["name"] or "Unknown filename"
+        describe_files(metadata)
+        downloaded_file = await self.download(metadata)
+        if not downloaded_file:
```

Comment:
Is the extractor still working? Your download function isn't returning anything, it seems. That means that this
`if not download_file:` would be true always, so the code below would unreachable.

Reply:
Fixed